File: server/src/models/Project.model.ts
import mongoose, { Schema } from 'mongoose';

const projectSchema = new Schema(
  {
    workspace: { type: Schema.Types.ObjectId, ref: 'Workspace', required: true },
    name: { type: String, required: true },
    clientName: { type: String, default: '' },
    description: { type: String, default: '' },
    status: { type: String, default: 'planning' },
    priority: { type: String, default: 'medium' },
    originalScope: { type: String, default: '' },
    deadline: { type: Date },
    createdBy: { type: Schema.Types.ObjectId, ref: 'User' },
  },
  { timestamps: true }
);

export const ProjectModel = mongoose.model('Project', projectSchema);

File: server-go/internal/modules/project/project_model.go
package project

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Project struct {
	ID            primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
	Workspace     primitive.ObjectID `bson:"workspace" json:"workspace"`
	Name          string             `bson:"name" json:"name"`
	ClientName    string             `bson:"clientName" json:"clientName"`
	Description   string             `bson:"description" json:"description"`
	Status        string             `bson:"status" json:"status"`
	Priority      string             `bson:"priority" json:"priority"`
	OriginalScope string             `bson:"originalScope" json:"originalScope"`
	Deadline      *time.Time         `bson:"deadline,omitempty" json:"deadline,omitempty"`
	CreatedBy     primitive.ObjectID `bson:"createdBy" json:"createdBy"`
	CreatedAt     time.Time          `bson:"createdAt" json:"createdAt"`
	UpdatedAt     time.Time          `bson:"updatedAt" json:"updatedAt"`
}
